static/app/components/tabs/tab.tsx

```diff
@@ -0,0 +1,178 @@
+import {forwardRef} from 'react';
+import styled from '@emotion/styled';
+import {useTab} from '@react-aria/tabs';
+import {mergeProps, useObjectRef} from '@react-aria/utils';
+import {TabListState} from '@react-stately/tabs';
+import {Node, Orientation} from '@react-types/shared';
+
+import space from 'sentry/styles/space';
+
+interface TabProps<T> {
+  item: Node<T>;
+  orientation: Orientation;
+  overflowing: boolean;
+  state: TabListState<T>;
+}
+
+/**
+ * Renders a single tab item. This should not be imported directly into any
+ * page/view – it's only meant to be used by <TabsList />. See the correct
+ * usage in tabs.stories.js
+ */
+function BaseTab<T>(
+  {item, state, orientation, overflowing}: TabProps<T>,
+  forwardedRef: React.ForwardedRef<HTMLLIElement>
+) {
+  const ref = useObjectRef(forwardedRef);
+
+  const {key, rendered} = item;
+  const {tabProps, isSelected, isDisabled} = useTab({key}, state, ref);
+
+  return (
+    <TabWrap
+      {...mergeProps(tabProps)}
+      disabled={isDisabled}
+      selected={isSelected}
+      overflowing={overflowing}
+      orientation={orientation}
+      ref={ref}
+    >
+      <HoverLayer orientation={orientation} />
+      <FocusLayer orientation={orientation} />
+      {rendered}
+      <TabSelectionIndicator orientation={orientation} selected={isSelected} />
+    </TabWrap>
+  );
+}
+
+export const Tab = forwardRef(BaseTab);
+
+const TabWrap = styled('li')<{
+  disabled: boolean;
+  orientation: Orientation;
+  overflowing: boolean;
+  selected: boolean;
+}>`
+  display: flex;
+  align-items: center;
+  position: relative;
+  height: calc(
+    ${p => p.theme.form.sm.height}px +
```

Comment:
Why form?

Reply:
We use `form` styles for buttons (in addition to form elements). I'm using it here to make sure that the tab items have the same height as the overflow trigger button:
<img width="162" alt="Screen Shot 2022-09-19 at 2 49 21 PM" src="https://user-images.githubusercontent.com/44172267/191125832-5aa8afa1-502c-4645-9d64-dc4488a6ceb8.png">

Reply:
Got you